File: ecoquest/lib/model/mission.dart
class Mission {
  final String id;
  final String titulo;
  final String descricao;
  final String icon;
  final int quantidadeAtual;
  final int quantidadeTotal;
  final String nivelDificuldade;
  final double experience;
  final double points;

 factory Mission.fromJson(Map<String, dynamic> json) {
    return Mission(
      id: json['id'],
      titulo: json['titulo'],
      descricao: json['descricao'],
      icon: json['icon'],
      quantidadeAtual: json['quantidadeAtual'],
      quantidadeTotal: json['quantidadeTotal'],
      nivelDificuldade: json['nivelDificuldade'],
      experience: json['experience'],
      points: json['points'],
    );
  }


  const Mission({
    required this.id,
    required this.titulo,
    required this.descricao,
    required this.icon,
    required this.quantidadeAtual,
    required this.quantidadeTotal,
    required this.nivelDificuldade,
    required this.experience,
    required this.points
  });
}

List<Mission> parseMissoes(List<Map<String, dynamic>> jsonList) {
  return jsonList.map((data) => Mission.fromJson(data)).toList();
}

File: ecoquest_api/src/main/java/com/eco/quest/api/ecoquest_api/model/Mission.java
package com.eco.quest.api.ecoquest_api.model;

import java.math.BigDecimal;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
@Entity
public class Mission {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;
    private String titulo;
    private String descricao;
    private String icon;
    private Long quantidadeAtual;
    private Long quantidadeTotal;
    private Dificuldade nivelDificuldade;
    private BigDecimal experience;
    private BigDecimal points;
    @ManyToOne
    private Categoria categoria;

    
}
